spec/support/i18n_cov.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "singleton"
+
+class I18nCov
+  include Singleton
+
+  attr_reader :locales, :used_keys
+
+  module I18nOverrides
+    def lookup(locale, key, scope = [], options = {})
+      I18nCov.log(key, scope)
+      super
+    end
+  end
+
+  class << self
+    delegate :start, :report, :log, to: :instance
+  end
+
+  def start(locales = nil)
+    @used_keys = []
+    @locales = locales || get_in_app_locales
+    I18n::Backend::Simple.include I18nOverrides
+  end
+
+  def log(key, scope)
+    used_keys << (Array(scope || []) + [key]).compact.join(".")
+  end
+
+  def report
+    used = used_keys.uniq
+    all = get_all_i18n_keys
+    unused = all.reject { |as| used.any? { |us| as.start_with?(us) } }
+
+    if unused.any?
+      write_report(unused)
+      print_message(all, unused)
+    end
+  end
+
+private
+
+  def get_in_app_locales
+    ymls = Dir.glob(Rails.root.join("config", "locales", "**", "*.yml"))
+    ymls.map { |yml| YAML.load_file(yml) }.reduce({}, :deep_merge)
+  end
+
+  def get_all_i18n_keys
+    locales.flat_map { |locale, ts| key_chains_for(locale, ts) }
+      .map { |key_chain| key_chain.drop(1).join(".") }
```

Comment:
This might need some documentation - refactor, comment, or commit message - to explain the drop of the locale part